pkg/api/v1/router.go

```diff
@@ -10,49 +10,99 @@ import (
-
-	rg.GET("/server-component-types", amw.AuthRequired(), amw.RequiredScopes(readScopes("server-component-types")), r.serverComponentTypeList)
-	rg.POST("/server-component-types", amw.AuthRequired(), amw.RequiredScopes(updateScopes("server-component-types")), r.serverComponentTypeCreate)
-
-	rg.GET("/server-component-firmwares", amw.AuthRequired(), amw.RequiredScopes(readScopes("server")), r.serverComponentFirmwareList)
-	rg.POST("/server-component-firmwares", amw.AuthRequired(), amw.RequiredScopes(createScopes("server")), r.serverComponentFirmwareCreate)
-	rg.GET("/server-component-firmwares/:uuid", amw.AuthRequired(), amw.RequiredScopes(readScopes("server")), r.serverComponentFirmwareGet)
-	rg.PUT("/server-component-firmwares/:uuid", amw.AuthRequired(), amw.RequiredScopes(updateScopes("server")), r.serverComponentFirmwareUpdate)
-	rg.DELETE("/server-component-firmwares/:uuid", amw.AuthRequired(), amw.RequiredScopes(deleteScopes("server")), r.serverComponentFirmwareDelete)
+	// require all calls to have auth
+	rg.Use(amw.AuthRequired())
+
+	// /servers
+	srvs := rg.Group("/servers")
+	{
+		srvs.GET("", amw.RequiredScopes(readScopes("server")), r.serverList)
+		srvs.POST("", amw.RequiredScopes(createScopes("server")), r.serverCreate)
+
+		srvs.GET("/components", amw.RequiredScopes(readScopes("server:component")), r.serverComponentList)
+
+		// /servers/:uuid
+		srv := srvs.Group("/:uuid")
+		{
+			srv.GET("", amw.RequiredScopes(readScopes("server")), r.serverGet)
+			srv.PUT("", amw.RequiredScopes(updateScopes("server")), r.serverUpdate)
+			srv.DELETE("", amw.RequiredScopes(deleteScopes("server")), r.serverDelete)
+
+			// /servers/:uuid/attributes
+			srvAttrs := srv.Group("/attributes")
+			{
+				srvAttrs.GET("", amw.RequiredScopes(readScopes("server", "server:attributes")), r.serverAttributesList)
+				srvAttrs.POST("", amw.RequiredScopes(createScopes("server", "server:attributes")), r.serverAttributesCreate)
+				srvAttrs.GET("/:namespace", amw.RequiredScopes(readScopes("server", "server:attributes")), r.serverAttributesGet)
+				srvAttrs.PUT("/:namespace", amw.RequiredScopes(updateScopes("server", "server:attributes")), r.serverAttributesUpdate)
+				srvAttrs.DELETE("/:namespace", amw.RequiredScopes(deleteScopes("server", "server:attributes")), r.serverAttributesDelete)
+			}
+
+			// /servers/:uuid/components
+			srvComponents := srv.Group("/components")
+			{
+				srvComponents.POST("", amw.RequiredScopes(createScopes("server", "server:component")), r.serverComponentsCreate)
+				srvComponents.GET("", amw.RequiredScopes(readScopes("server", "server:component")), r.serverComponentGet)
+				srvComponents.PUT("", amw.RequiredScopes(updateScopes("server", "server:component")), r.serverComponentUpdate)
+			}
+
+			// /servers/:uuid/secrets/:slug
+			svrSecret := srv.Group("secrets/:slug")
```

Comment:
I appreciate that you need a specific scope now to read and write secrets; which are separate than the regular `read` and `write` scopes.